tests/protocol.rs

```diff
@@ -90,11 +97,11 @@ async fn simulate_mpc_async(
             }
             outputs.retain(|o| !o.is_empty());
```

Comment:
This retain is wrong. Currently, `outputs` will contain empty vectors if `mpc` failed with an error (line 70). If we remove these empty vecs then, we might not notice that an error occurred for one of the parties.